scripts/test_translator.py

```diff
@@ -244,11 +244,20 @@ def __init__(self, full_path: str, files: 're.Pattern', keep: List[str], log_lev
         # are broken without it on macOS 12.
         # limit this to macOS because if we do happen to have multiple versions of Clang around, we
         # don't know which to use here, and using the wrong can one break things badly
-        if sys.platform == "darwin":
+        if sys.platform == "darwin" or os.environ.get("C2RUST_USE_NIX") == "1":
             _, stdout, _ = clang["-print-resource-dir"].run(retcode=None)
-            self.clang_resource_dir = " \"-I{}/include\",".format(stdout.strip())
+            self.clang_extra_flags = " \"-I{}/include\",".format(stdout.strip())
         else:
-            self.clang_resource_dir = ""
+            self.clang_extra_flags = ""
```

Comment:
We lose the naming of `clang_resource_dir` here, so could you leave it as a `clang_resource_dir` local variable here for this `if`/`else` and then add a separate `self.clang_extra_flags = clang_resource_dir + nix_compile_flags` line after determining each of them?

Reply:
Okay, I've made this change (well, I just do `self.clang_extra_flags += nix_compile_flags` in the nix if statement. I'm happy to make `nix_compile_flags` an empty string in a branch or something instead, but I thought it might be more clear that you're not getting weird nix flags unless you're actually using nix)

Reply:
This seems good, thanks!